### DISCUSSION TITLE
Calculation of MCSS is taking long time for particular molecules

### DISCUSSION THREAD

**Discussion by zmahnoor14** [rdkit/rdkit#5582] [Q&A] [answered]
Dear RDKit community,

Thank you for the amazing tool.

I am currently working on calculating MCSS with these two SMILES.

i = C1C(C(OC2=CC(=CC(=C21)O)O)C3=CC4=C(C=C3)OC5(C(O4)(C(=O)C6=C(C=C(C=C6O5)O)O)O)C7=CC(=C(C=C7)O)O)O
j = C1C(C(OC2=CC(=CC(=C21)O)O)C3=CC4=C(C=C3)OC5(C(=O)C6=C(C=C(C=C6OC5(O4)C7=CC(=C(C=C7)O)O)O)O)O)O

I run the code, but it keeps on running with default settings for hours. A similar issue has been previously submitted.

[Link to previous issue](https://github.com/rdkit/rdkit/discussions/4725)

Using version 2022.03.5, the code is:
```python
n = [Chem.MolFromSmiles(i),
    Chem.MolFromSmiles(j)]

res = rdFMCS.FindMCS(n)
```
Kind regards,
Mahnoor

**Comment by vfscalfani** [comment, marked as answer]
Hi @zmahnoor14

Maybe it would be useful to set a timeout to return the best match? Here is a quote from the RDKit Getting Started Guide:

https://www.rdkit.org/docs/GettingStartedInPython.html

> The MCS algorithm will exhaustively search for a maximum common substructure. Typically this takes a fraction of a second, but for some comparisons this can take minutes or longer. Use the timeout parameter to stop the search after the given number of seconds (wall-clock seconds, not CPU seconds) and return the best match found in that time. If timeout is reached then the canceled property of the MCSResult will be True instead of False.

Tagging #4725 related question here.
